MobileWallet/Common/NotificationManager.swift

```diff
@@ -95,4 +155,220 @@ class NotificationManager {
+        }
+
+        guard let signature = try? wallet.signMessage("\(fromPubKeyHex)\(toPubKeyHex)") else {
+            return TariLogger.error("Failed to sign message")
+        }
+
+        let url = URL(string: "\(TariSettings.shared.pushNotificationServer)/send/\(toPubKeyHex)")!
+
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.setValue("application/json", forHTTPHeaderField: "Accept")
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.httpBody = try JSONEncoder().encode(
+            SendNotificationServerRequest(
+                from_pub_key: fromPubKeyHex,
+                signature: signature.hex,
+                public_nonce: signature.nonce
+            )
+        )
+
+        let onRequestError = {(error: Error) in
+            onError(error)
+            NotificationManager.isSendRequestInProgress = false
+        }
+
+        NotificationManager.isSendRequestInProgress = true
+
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+            guard error == nil else {
+                onRequestError(error!)
+                return
+            }
+
+            guard let data = data, let response = response as? HTTPURLResponse else {
+                onRequestError(TokenServerError.unknown)
+                return
+            }
+
+            guard response.statusCode != 403 else {
+                onRequestError(TokenServerError.invalidSignature)
+                return
+            }
+
+            var responseDict: [String: Any]?
+            if let responseString = String(data: data, encoding: .utf8) {
+                if let data = responseString.data(using: .utf8) {
+                    do {
+                        responseDict = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
+                    } catch {
+                        onRequestError(error)
+                        return
+                    }
+                }
+            }
+
+            guard (200 ... 299) ~= response.statusCode else {
+                var message: String?
+                if let res = responseDict {
+                    message = res["error"] as? String
+                }
+
+                onRequestError(TokenServerError.server(response.statusCode, message: message))
+                return
+            }
+
+            guard let res = responseDict else {
+                onRequestError(TokenServerError.missingResponse)
+                return
+            }
+
+            guard let sent = res["sent"] as? Bool else {
+                onRequestError(TokenServerError.responseInvalid)
+                return
+            }
+
+            if sent {
+                onSuccess()
+                NotificationManager.isSendRequestInProgress = false
+            } else {
+                onRequestError(TokenServerError.pushNotSent)
+            }
+        }
+
+        task.resume()
```

Comment:
`sendToRecipient` function body looks similar to `registerDeviceToken`, might be some space for reuse around here for future refactoring.

Reply:
Ok let me do this with the rest of the notification UX, I need to reuse the `sendToRecipient`  to also work for when a recipient notifies a sender again